internal/storage/clickhouse.go

```diff
@@ -306,36 +307,88 @@ func executeQuery[T any](c *ClickHouseConnector, table, columns string, qf Query
 func (c *ClickHouseConnector) buildQuery(table, columns string, qf QueryFilter) string {
 	query := fmt.Sprintf("SELECT %s FROM %s.%s FINAL WHERE is_deleted = 0", columns, c.cfg.Database, table)
 
-	if qf.ContractAddress != "" {
-		query += fmt.Sprintf(" AND address = '%s'", qf.ContractAddress)
-	}
+	query = addToAddress(table, query, qf.ToAddress)
+
+	// Add signature clause
 	if qf.Signature != "" {
 		query += fmt.Sprintf(" AND topic_0 = '%s'", qf.Signature)
 	}
+	// Add filter params
 	for key, value := range qf.FilterParams {
-		query += fmt.Sprintf(" AND %s = '%s'", key, value)
+		query = addFilterParams(key, strings.ToLower(value), query)
 	}
 
+	// Add sort by clause
 	if qf.SortBy != "" {
 		query += fmt.Sprintf(" ORDER BY %s %s", qf.SortBy, qf.SortOrder)
 	}
 
+	// Add limit clause
 	if qf.Page > 0 && qf.Limit > 0 {
 		offset := (qf.Page - 1) * qf.Limit
 		query += fmt.Sprintf(" LIMIT %d OFFSET %d", qf.Limit, offset)
 	} else {
+		// Add limit clause
 		query += getLimitClause(int(qf.Limit))
 	}
 
 	return query
 }
 
+func addFilterParams(key, value, query string) string {
+	// if the key includes topic_0, topic_1, topic_2, topic_3, apply left padding to the value
+	if strings.Contains(key, "topic_") {
+		value = getTopicValueFormat(value)
+	}
+
+	suffix := key[len(key)-3:]
+	switch suffix {
+		case "gte":
+			query += fmt.Sprintf(" AND %s >= '%s'", key[:len(key)-3], value)
+		case "lte":
+			query += fmt.Sprintf(" AND %s <= '%s'", key[:len(key)-3], value)
+		case "_lt":
+			query += fmt.Sprintf(" AND %s < '%s'", key[:len(key)-3], value)
+		case "_gt":
+			query += fmt.Sprintf(" AND %s > '%s'", key[:len(key)-3], value)
+		case "_ne":
+			query += fmt.Sprintf(" AND %s != '%s'", key[:len(key)-3], value)
+		case "_in":
+			query += fmt.Sprintf(" AND %s IN (%s)", key[:len(key)-3], value)
+		default:
+			query += fmt.Sprintf(" AND %s = '%s'", key, value)
+	}
+	return query
+}
+
+func addToAddress(table, query string, toAddress string) string {
+	toAddress = strings.ToLower(toAddress)
+	// This needs to move to a query param that accept multiple addresses
+	if table == "logs" {
+		if toAddress != "" {
+			query += fmt.Sprintf(" AND address = '%s'", toAddress)
+		}
```

Comment:
What's the intended behaviour here?
`address` is the address of the contract that emitted the log. If you are thinking about a transaction, then not all of the logs for that transaction will be there if you search by `to`

Reply:
yes we have the "to" address in transactions and and the address in events. I agree naming them the same is confusing. Im changing the name to contract for events